Added `JsonResponse` (#6174)
scrapy/scrapy: docs/topics/request-response.rst

    line.  See :attr:`TextResponse.encoding`.

.. _bug in lxml: https://bugs.launchpad.net/lxml/+bug/1665241

JsonResponse objects
--------------------

.. class:: JsonResponse(url[, ...])

    The :class:`JsonResponse` class is a subclass of :class:`TextResponse` which
    adds encoding auto-discovering support by looking into the JSON declaration

>     line.  See :attr:`TextResponse.encoding`.

Reviewer: I don't think this makes sense?

Author: Oh you're right, you think this might work:

```rst
    The :class:`JsonResponse` class is a subclass of :class:`TextResponse` which
    adds encoding auto-discovering support by looking for a MIME type in the
    `Content-Type` header that represent to a JSON object.  See :attr:`TextResponse.encoding`.
```

>     The :class:`JsonResponse` class is a subclass of :class:`TextResponse` which
>     adds encoding auto-discovering support by looking for a MIME type in the
>     `Content-Type` header that represent to a JSON object.  See :attr:`TextResponse.encoding`.

Reviewer 2: I think @wRAR’s point is that “adds encoding auto-discovering support” seems incorrect or misleading.

Something like this is probably more accurate:

```suggestion
    The :class:`JsonResponse` class is a subclass of :class:`TextResponse` 
    that is used when the response has a `JSON MIME type 
    <https://mimesniff.spec.whatwg.org/#json-mime-type>`_ in its `Content-Type` 
    header.
```

Although I am unsure of whether or not this change, i.e. #6069, is worth implementing.

Author: Oh I see. Thanks
